handshake/crypto_setup_tls.go

```diff
@@ -0,0 +1,140 @@
+package handshake
+
+import (
+	"crypto/tls"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/bifurcation/mint"
+	"github.com/lucas-clemente/quic-go/crypto"
+	"github.com/lucas-clemente/quic-go/internal/protocol"
+	"github.com/lucas-clemente/quic-go/internal/utils"
+)
+
+// KeyDerivationFunction is used for key derivation
+type KeyDerivationFunction func(crypto.MintState, protocol.Perspective) (crypto.AEAD, error)
+
+type cryptoSetupServerTLS struct {
+	mutex sync.RWMutex
+
+	perspective protocol.Perspective
+
+	keyDerivation KeyDerivationFunction
+
+	mintConf *mint.Config
+	conn     *mint.Conn
+
+	nullAEAD crypto.AEAD
+	aead     crypto.AEAD
+
+	aeadChanged chan<- protocol.EncryptionLevel
+}
+
+// NewCryptoSetupTLS creates a new CryptoSetup instance for a server
+func NewCryptoSetupTLS(
+	hostname string, // only needed for the client
+	perspective protocol.Perspective,
+	tlsConfig *tls.Config,
+	cryptoStream io.ReadWriter,
+	aeadChanged chan<- protocol.EncryptionLevel,
+) (CryptoSetup, error) {
+	mintConf, err := tlsToMintConfig(tlsConfig, perspective)
+	if err != nil {
+		return nil, err
+	}
+	mintConf.ServerName = hostname
+	var conn *mint.Conn
+	if perspective == protocol.PerspectiveServer {
+		conn = mint.Server(&fakeConn{cryptoStream}, mintConf)
+	} else {
+		conn = mint.Client(&fakeConn{cryptoStream}, mintConf)
+	}
+	return &cryptoSetupServerTLS{
+		perspective:   perspective,
+		mintConf:      mintConf,
+		conn:          conn,
+		nullAEAD:      crypto.NewNullAEAD(perspective, protocol.Version39),
```

Comment:
Why is there a version const here?

Reply:
I think I forgot to write the correct null AEAD constructor. I will add this later today.

Reply:
Fixed.